src/wallet/wallet2.cpp

```diff
@@ -245,8 +247,16 @@ std::unique_ptr<tools::wallet2> make_basic(const boost::program_options::variabl
 		daemon_port = config_get_rpc_port(nettype);
 	}
 
-	if(daemon_address.empty())
+	if(!daemon_address.empty())
+	{
+		std::regex valid_address(".+:[0-9]+");
```

Comment:
could you please use `".+:[0-9]{1,5}"` to enforce that the port has 1 to 5 digits